src/canvas/CanvasOptions.ts

```diff
@@ -0,0 +1,296 @@
+import type { ModifierKey, TOptionalModifierKey } from '../EventTypeDefs';
+import type { TOptions } from '../typedefs';
+import type { StaticCanvasOptions } from './StaticCanvasOptions';
+
+export interface CanvasTransformOptions {
+  /**
+   * When true, objects can be transformed by one side (unproportionally)
+   * when dragged on the corners that normally would not do that.
+   * @type Boolean
+   * @default
+   * @since fabric 4.0 // changed name and default value
+   */
+  uniformScaling: boolean;
```

Comment:
In retrospective i don't like the idea of moving the documentation out of the classes themselves. I understand it makes files shorter but it seems the wrong solution.
\I know we did this already and is not new, but now i really dislike it.

Opening a class and being able to read is important to me

Reply:
If you hover over a prop it shows the jsdoc.
I like it but that is why I kept it in separate commits... because I wanted to give you an easy way out.

Reply:
no if we go out we change them all. For now we keep consistency.
I know hovering works but reading is reading. It is also a vscode thing not an universal one

Reply:
So should I revert or not?

Reply:
Should the jsdoc exist on the options as well?
For a dev the important place is on options so when creating a canvas ir object they see what it can do.

Reply:
I agree with @asturur, better a big module with everything related than death-by-thousand-files. While vscode hover documentation helps, there are many other occasions where it's useful to have things colocated in the same file, e.g. PR reviewing or debugging.

Reply:
I'd recommend this book where the topic of "deep" modules is explained: https://www.amazon.com/Philosophy-Software-Design-John-Ousterhout/dp/1732102201, i.e. a **deep** module with the relevant implementations all together (as far as the exposed public API is small) is better than many **shallow** modules that you need to put together in your mind in order to understand.